src/Raven.Studio/typescript/viewmodels/manage/trafficWatch.ts

```diff
@@ -72,13 +90,42 @@ class trafficWatch extends viewModelBase {
     private matchesFilter(item: Raven.Client.Documents.Changes.TrafficWatchChange) {
         const filter = this.filter();
         if (!filter) {
-            return true;
+            return this.matchFilteredTypeData(item);
         }
         const filterLowered = filter.toLocaleLowerCase();
         const uri = item.RequestUri.toLocaleLowerCase();
-        return uri.includes(filterLowered);
+        return uri.includes(filterLowered) && this.matchFilteredTypeData(item);
     }
-    
+
+    private matchFilteredTypeData(item: Raven.Client.Documents.Changes.TrafficWatchChange) {
+        return this.filteredTypeData.find(x => x.propertyName === item.Type && x.isChecked());
+    }
+
+    filterOnClick(data: typeData):  void {
+        if (data.isChecked() === false) {
```

Comment:
can we reverse logic to avoid if (something === false) { doFirst() } else { doSecond} ?)
Instead we can do: if (something) { doSecond() } else { doFirst}

Reply:
done